x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -282,49 +281,49 @@ func (c *runCmd) createCallParams(ctx context.Context, db state.Immutable, param
 			if err != nil {
 				return nil, err
 			}
-			cp = append(cp, runtime.CallParam{Value: val})
+			cp = append(cp, program.CallParam{Value: val})
 		case Bool:
 			val, ok := param.Value.(bool)
 			if !ok {
 				return nil, fmt.Errorf("%w: %s", ErrFailedParamTypeCast, param.Type)
 			}
-			cp = append(cp, runtime.CallParam{Value: boolToUint64(val)})
+			cp = append(cp, program.CallParam{Value: boolToUint64(val)})
 		case KeyEd25519: // TODO: support secp256k1
 			val, ok := param.Value.(string)
 			if !ok {
 				return nil, fmt.Errorf("%w: %s", ErrFailedParamTypeCast, param.Type)
 			}
 
-			var key string
+			key := val
 			// get named public key from db
 			pk, ok, err := storage.GetPublicKey(ctx, db, val)
 			if !ok {
```

Comment:
since we are setting `key := val` above we can change the if-else statement to just
```go
if ok {			
  key = string(pk[:])
}
```